crates/miden-lib/src/note/well_known_note.rs

```diff
@@ -157,4 +161,145 @@ impl WellKnownNote {
             },
         }
     }
+
+    /// Checks the correctness of the provided note inputs against the target account.
+    ///
+    /// It returns the corresponding note consumption status in case we can guarantee that the note
+    /// cannot be consumed, or `None` otherwise.
+    ///
+    /// It performs:
+    /// - for all notes: a
+    /// - for `P2ID` note:
+    ///     - check that note inputs have correct number of values.
+    ///     - assertion that the account ID provided by the note inputs is equal to the target
+    ///       account ID.
+    /// - for `P2IDE` note:
+    ///     - check that note inputs have correct number of values.
+    ///     - assertion that the timelock block height is reached, so the note can be consumed.
+    ///     - assertion that the account ID provided by the note inputs is equal to the target
+    ///       account ID (which means that the note is going to be consumed by the target account)
+    ///       or that the target account ID is equal to the sender account ID (which means that the
+    ///       note is going to be consumed by the sender account)
+    pub fn check_note_inputs(
+        &self,
+        note: &Note,
+        target_account_id: AccountId,
+        block_ref: BlockNumber,
+    ) -> Option<NoteConsumptionStatus> {
+        match self {
+            WellKnownNote::P2ID => {
+                let note_inputs = note.inputs().values();
+                if note_inputs.len() != self.num_expected_inputs() {
+                    return Some(NoteConsumptionStatus::Incompatible(format!(
+                        "P2ID note should have 2 inputs, but {} was provided",
+                        note_inputs.len()
+                    )));
+                }
+
+                let Some(input_account_id) = try_read_account_id_from_inputs(note_inputs) else {
+                    return Some(NoteConsumptionStatus::Incompatible(
+                        "Account ID provided to the P2ID note inputs is invalid".to_string(),
+                    ));
+                };
+
+                if input_account_id == target_account_id {
+                    None
+                } else {
+                    Some(NoteConsumptionStatus::Incompatible("Account ID provided to the P2ID note inputs doesn't match the target account ID".to_string()))
+                }
+            },
+            WellKnownNote::P2IDE => {
+                let note_inputs = note.inputs().values();
+                if note_inputs.len() != self.num_expected_inputs() {
+                    return Some(NoteConsumptionStatus::Incompatible(format!(
+                        "P2IDE note should have 4 inputs, but {} was provided",
+                        note_inputs.len()
+                    )));
```

Comment:
Nit: Should we format in the `num_expected_inputs` instead of hardcoding `4`, so this does not go stale when we change P2IDE?